include/zephyr/device.h

```diff
@@ -887,6 +893,104 @@ __syscall int device_init(const struct device *dev);
 
 #endif /* CONFIG_DEVICE_DEPS */
 
+#if defined(CONFIG_DEVICE_DT_METADATA) || defined(__DOXYGEN__)
+/**
+ * @brief Devicetree node labels associated with a device
+ */
+struct device_dt_nodelabels {
+	/* @brief number of elements in the nodelabels array */
+	size_t num_nodelabels;
+	/* @brief array of node labels as strings, exactly as they
+	 *        appear in the final devicetree
+	 */
+	const char *nodelabels[];
```

Comment:
Flexible array members are part of C99. Was this tested on multiple toolchains?

Reply:
We can assume C99 anywhere in Zephyr, as far as I know.

> Zephyr does not target a specific version of the C standards; however, the Zephyr codebase makes extensive use of the features newly introduced in the 1999 release of the ISO C standard (ISO/IEC 9899:1999, hereinafter referred to as C99) such as those listed below, effectively requiring the use of a compiler toolchain that supports the C99 standard and above

https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/develop/languages/c/index.rst#language-standards

Reply:
To answer your question directly, though: no, I only tested with Zephyr SDK.